_overviews/scala3-book/methods-most.md

```diff
@@ -0,0 +1,396 @@
+---
+title: Method Features
+type: section
+description: This section introduces Scala 3 methods, including main methods, extension methods, and more.
+num: 24
+previous-page: methods-intro
+next-page: methods-main-methods
+---
+
+
+
+In Scala 2, *methods* can be defined inside classes, traits, objects, case classes, and case objects. But it gets better: In Scala 3 they can also be defined _outside_ any of those constructs with a new feature named Toplevel definitions. In short, methods can now be defined anywhere.
+
+This section demonstrates many of the features of defining and calling methods.
+
+
+
+## Defining methods
+
+Scala methods have many features, including these:
+
+- Generic parameters
+- `using` parameters
+- Default parameter values
+- Multiple parameter groups
+- By-name parameters
+- Function parameters
+- Potentially more ...
+
+Some of these features are demonstrated later in this section, but when you’re defining a “simple” method that doesn’t use those features, the syntax looks like this:
+
+```scala
+def methodName(param1: Type1, param2: Type2): ReturnType = 
+  // the method body
+  // goes here
+end methodName   // this is optional
+```
+
+In that syntax:
+
+- The keyword `def` is used to define a method
+- The Scala standard is to name methods using the camel case convention
+- Method parameters are always defined with their type
+- Declaring the method return type is optional
+- Methods can consist of many lines, or just one line
+- Providing the `end methodName` portion after the method body is also optional, and is only recommended for long methods
+
+Here are two examples of a one-line method named `add` that takes two `Int` input parameters. The first version explicitly shows the method’s `Int` return type, and the second does not:
+
+```scala
+def add(a: Int, b: Int): Int = a + b
+def add(a: Int, b: Int) = a + b
+```
+
+You can use either style. For one-line methods it can be nice to leave the return type off, if you prefer. For larger methods, declaring the return type can make it easier to understand the method when you look at it months or years later, or when you look at another person’s methods.
```

Comment:
```suggestion
You can use either style. For one-line methods it can be nice to leave the return type off, if you prefer. For larger methods, declaring the return type can make it easier to understand it when you look at it months or years later, or when you look at another person’s code.
```